File: Data Structures & Algorithms/maximum-subarray/submission-1.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Kadane's Algorithm
        // Two points and temp

        int maxSum = nums[0];
        int currSum = 0;

        for(int i=0; i<nums.length; i++){
            currSum = Math.max(nums[i], nums[i] + currSum);
            maxSum = Math.max(maxSum, currSum);
        }
        return maxSum;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-1.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        n = len(nums)
        L = 0
        max_freq = 0

        for R in range(n):
            
            if(nums[R] == 0):
                if(R < n):
                    L = R + 1
            max_freq = max(max_freq, R - L + 1)

        # [1,0,1,1,0,0]
        #            R
        #            L
        return max_freq

